internal/configconverter/loglevel_to_verbosity.go

```diff
@@ -0,0 +1,93 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package configconverter
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"regexp"
+
+	"go.opentelemetry.io/collector/config/configtelemetry"
+	"go.opentelemetry.io/collector/confmap"
+	"go.uber.org/zap/zapcore"
+)
+
+type LogLevelToVerbosity struct{}
+
+func (LogLevelToVerbosity) Convert(_ context.Context, in *confmap.Conf) error {
+	if in == nil {
+		return fmt.Errorf("cannot MoveHecTLS on nil *confmap.Conf")
+	}
+
+	const expression = "exporters::logging(/\\w+)?::loglevel"
+	re := regexp.MustCompile(expression)
+	out := map[string]any{}
+	unsupportedKeyFound := false
+	for _, k := range in.AllKeys() {
+		v := in.Get(k)
+		match := re.FindStringSubmatch(k)
+		if match == nil {
+			out[k] = v
+		} else {
+			// check if verbosity is also set:
+			verbosityKey := fmt.Sprintf("exporters::logging%s::verbosity", match[1])
+			if in.Get(verbosityKey) == nil {
+				log.Printf("Deprecated key found: %s. Translating to %s\n", k, verbosityKey)
+				var l zapcore.Level
```

Comment:
Why would it be necessary to use zap in addition to configtelemetry?

Reply:
the loglevel is a string representation of a zapcore.Level, and we have to translate it over to verbosity levels using the logic from the logging exporter config.go file: https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/loggingexporter/config.go#L48

Reply:
right, but that's in the exporter itself to relay to a new zap instance, where we are only obligated to work with the config value itself. Why validate it at all and not transparently forward?

Reply:
Oh my bad, I forgot the field type changed in total and you're coercing it to the intermediary.